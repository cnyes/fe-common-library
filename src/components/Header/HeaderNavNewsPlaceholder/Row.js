import React from 'react';
import styles from './Row.scss';

const Row = () => {
  return (
    <div className={styles['multi-row']}>
      <article className={styles['news-row-placeholder']}>
        <div className={styles.wrapper} />
      </article>
      <article className={styles['news-row-placeholder']}>
        <div className={styles.wrapper} />
      </article>
      <article className={styles['news-row-placeholder']}>
        <div className={styles.wrapper} />
      </article>
    </div>
  );
};

export default Row;
