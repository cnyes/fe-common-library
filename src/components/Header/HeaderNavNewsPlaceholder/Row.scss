@import '../../../styles/base';

@keyframes float_bg {
  from { background-position: 100% 0; }
  to { background-position: -100% 0; }
}

.multi-row {
  width: 100%;
}

.news-row-placeholder {
  display: block;
  position: relative;
  background: $white;
  height: 36px;
  border-bottom: 1px solid $border-light-gray;
  width: 100%;
  padding: 0 20px;
  padding-left: 98px;

  .wrapper {
    display: block;
    position: absolute;
    left: 0;
    // animation: float_bg 1s infinite linear forwards;
    background: $gray-f6f7f8;
    background-image: -webkit-linear-gradient(left, $gray-f6f6f6 0%, $gray-edeef1 20%, $gray-f6f6f6 40%, $gray-f6f6f6 100%);
    background-repeat: no-repeat;
    background-size: 800px 100%;
    /* padding: 98px, cover width: 106px, margin to cover: 15px + 40px; */
    width: 100%;
    height: 15px;
    z-index: 2;
    @extend %vertical-center;
  }
}
