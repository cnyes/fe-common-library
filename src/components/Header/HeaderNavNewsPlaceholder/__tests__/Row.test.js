/* eslint-disable import/first,func-names,global-require,prefer-arrow-callback */
import React from 'react';
import { mount } from 'enzyme';
import Row from '../Row';

describe('Row', () => {
  it('should make a snapshot', () => {
    const subject = mount(<Row />);

    expect(subject.html()).toMatchSnapshot();
  });
});
