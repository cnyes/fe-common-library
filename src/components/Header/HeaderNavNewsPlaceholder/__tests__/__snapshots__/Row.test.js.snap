// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row should make a snapshot 1`] = `"<div class=\\"multi-row\\"><article class=\\"news-row-placeholder\\"><div class=\\"wrapper\\"></div></article><article class=\\"news-row-placeholder\\"><div class=\\"wrapper\\"></div></article><article class=\\"news-row-placeholder\\"><div class=\\"wrapper\\"></div></article></div>"`;
